src/sys/socket/sockopt.rs

```diff
@@ -513,23 +514,23 @@ unsafe impl<T> Get<CString> for GetCString<T> {
         &mut self.len
     }
 
-    unsafe fn unwrap(mut self) -> CString {
-        CStr::from_ptr(self.ffi_ptr() as *const c_char).to_owned()
+    unsafe fn unwrap(self) -> OsString {
```

Comment:
Does this method still need to be `unsafe`?

Reply:
You're right, but this is defined by the trait:
``` rust
unsafe trait Get<T> {
     ...
     unsafe fn unwrap(self) -> T;
}
```